src/main/java/seedu/financeit/goaltracker/GoalTracker.java

```diff
@@ -30,40 +31,103 @@
-        UiManager.printWithStatusIcon(Constants.PrintType.INSTRUCTION, "Input either expense/income "
-                + "to set goals for it."
-        );
-        backToMenu = false;
+        UiManager.printWithStatusIcon(Constants.PrintType.INSTRUCTION, "Input the following commands "
+                        + "to set goals for either expense or income.");
+        System.out.println("[1]. expense <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[2]. income <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[3]. To display goals for respective months, enter the following commands"
+                + " display expense/income for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[4]. edit expense/income <amount> for <Month in number form (e.g 01 -> Jan)>");
+        System.out.println("[5]. Enter <exit> to leave Goals Tracker");
+
         exitTracker = false;
         input = UiManager.handleInput();
-        cmdPacket = new InputParser().parseInput(input.toLowerCase());
+        cmdPacket = new InputParser().parseGoalCommand(input.toLowerCase());
+        editPacket = new InputParser().parseEditCommand(input.toLowerCase());
         while (!exitTracker) {
-            switch (cmdPacket.getCommandString()) {
+            switch (cmdPacket[0]) {
             case "expense":
                 setExpenseGoal();
                 break;
             case "income":
                 setIncomeGoal();
                 break;
+            case "display":
+                displayGoal();
+                break;
+            case "edit":
+                editGoal();
+                break;
             case "exit":
                 exitTracker = true;
                 break;
             default:
                 System.out.println("Invalid Command");
                 input = UiManager.handleInput();
-                cmdPacket = new InputParser().parseInput(input.toLowerCase());
+                cmdPacket = new InputParser().parseGoalCommand(input.toLowerCase());
                 break;
             }
         }
     }
 
+    public static void editGoal() {
+        try {
+            month = Month.of(Integer.parseInt(editPacket[4]));
+            if (editPacket[1].equals("expense")) {
+                expenseGoal = Integer.parseInt(editPacket[2]);
+                for (int i = 0; i < totalGoalList.getListSize(); i++) {
+                    try {
+                        if (totalGoalList.getGoal().get(i).getExpenseMonth().equals(month)) {
+                            goalSetter = new Goal(expenseGoal, "Expense", month);
+                            totalGoalList.getGoal().set(i, goalSetter);
+                            UiManager.printWithStatusIcon(Constants.PrintType.GOAL_STATUS, "You have successfully"
+                                    + " edited your expense goal for " + month);
+                            main();
```

Comment:
Can consider changing the totalGoalList to a hashMap with key {month} and value {goal}, since we only allow them to set 1 goal for the month.
E.g.
totalGoalList
JAN: 1500
FEB: 2000
MAR: 4500